include/zephyr/sys_clock.h

```diff
@@ -248,6 +248,24 @@ static inline bool sys_timepoint_expired(k_timepoint_t timepoint)
 	return K_TIMEOUT_EQ(sys_timepoint_timeout(timepoint), Z_TIMEOUT_NO_WAIT);
 }
 
+/**
+ * @brief Compare two timepoint values.
+ *
+ * @param a Timepoint to compare
+ * @param b Timepoint to compare against.
+ * @return true if a is is going to expire sooner than b, false otherwise.
```

Comment:
Nit: it can be possible that the time points are in the past.
```suggestion
 * @return true if a expires before b, false otherwise.
```